.github/workflows/push-tests.yml

```diff
@@ -9,9 +9,6 @@ on:
   pull_request:
     branches:
       - "*"
-  merge_group:
```

Comment:
this file contains the step that triggers a deployment run (see `create_deployment` step), which in turn is dependent on `push-prod-images` to run, which itself is dependent on the push tests passing. So this actually created a stall in the merge queue (when it was at the head of the queue, it blocked dev deployments queued after because it wasn't triggering push tests and GHA never reached the create_deployment step). 

I think its doable to remove the double push tests, but it would require per-step conditions (i.e. only run test steps in branch X, auto-pass them in branch Y to unblock create_deployment) or refactoring to move create_deployment to a different file

Reply:
Do you know how to do this, @nayib-jose-gloria ?

Reply:
I can take a look tomorrow!